### ISSUE TITLE
(Self-Voicing) Keyboard focus & reading blocks - design change to read automatically on focus

### ISSUE THREAD

**Issue by terracoda** [phetsims/gravity-force-lab-basics#220]
In an interview, highlighting and reading of reading blocks was a little buggy. On initial keyboard focus of the force block information, the block read aloud automatically on focus. The reading focus highlight appeared, but the highlighting did not.

Automatic reading on focus was not unexpected for the learner, and on subsequent visits on this block the leaner was surprised that the content did not voice automatically.

It may not be necessary to fix the behaviour, but to change the design.

Considering to make all content voice automatically on keyboard focus.

**Comment by terracoda** [comment]
I discussed this idea of self-voicing all content on keyboard focus with @emily-phet today, and we agreed that this change would make desirable alignments in the design and potentially be expected and beneficial for learners. This would also be very good for eventual non-visual access.

So, instead of figuring out why the current behavior is a little inconsistent, let's change the design so the reading blocks self-voice automatically on keyboard focus...AND that the block remains highlighted in yellow while self-voicing is happening.

**Comment by terracoda** [comment]
@jessegreenberg, assigning to you. Please re-assign if you have questions or if you need examples.

I am hoping this will make the keyboard interactions much less complicated, and even address the original concerns you had about handling content differently on keyboard focus.

The mouse and touch interactions remain the same - no change.

**Comment by jessegreenberg** [comment]
OK this is done in the above commits @terracoda. On slack I asked whether focus was now the ONLY way to reach this with keyboard or if enter/spacebar could still activate this content. I went with leaving spacebar and enter. So on initial focus you hear the content. Then you can hear the content again if you want to with keyboard commands.

The only thing I noticed was redundant content for the Mass controls in paradigm 2 - focus on the mass spinner reads the same content as focus on the whole panel. Not a big deal since I believe there is a strong preference for paradigm 3 these days.

**Comment by jessegreenberg** [comment]
I also noticed a visual issue with teh "R" icon. It becomes visible quickly as focus lands on things because the order of events is
- focus lands on object (so show the "speakable" rectangle with its "R" icon.
- Speaking starts after a delay from the browser (so show the yellow highlight around the content)
- Speaking ends (so the yellow highlight goes away and the "speakable" rectangle with its "R" icon comes back.

@terracoda if you would like a solution for this in this prototype please let me know.

**Comment by terracoda** [comment]
@jessegreenberg, we definitely should keep re-reading capability with enter/space interaction. Thanks for thinking of this!

Regarding what to do with the reading highlight. I need to investigate more.

**Comment by terracoda** [comment]
@jessegreenberg, hmm the highlights are a little inconsistent.
If I use the keyboard to access a reading block, then use the mouse and click to re-read the force arrow information, all visual highlights disappear.

This also happens if I am using the keyboard to access interactive objects and then use the mouse to read some content.
I am wondering if keyboard focus might be on something else and affecting what happens when I click?

I am not sure how to describe what's going on, it just seems a little if I interact quickly and switch input methods. This might not be typical behaviour.

**Comment by terracoda** [comment]
The reading automatically ion focus s working great, though.

**Comment by terracoda** [comment]
Discussed at Sep 2 meeting. @jessegreenberg to see if he can make some improvements to the consistency of when the highlights appear.

**Comment by terracoda** [comment]
@jessegreenberg, any update on this?

**Comment by terracoda** [comment]
I think this issue has been resolved, so closing.
